debugger/src/debug_adapter/dap_adapter.rs

```diff
@@ -1274,10 +1274,22 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
         } else {
             Some(memory_reference.saturating_add(byte_offset) as u32)
         };
+        // We can't rely on the MSDAP arguments to result in a memory aligned address for us to read from, so we force the read_pointer to be a memory_aligned address.
+        // TODO: Using 4 bytes for aligned memory reads is 32-Bit architecture specific, and needs to be addressed as part of 64-Bit implementation.
```

Comment:
I think this breaks with some Thumb 2 cases.  Thumb 2 instructions are 2-byte aligned, not 4.  They're also variable length of either 2 or 4 bytes, which means the simpler calculation at the top of computing the offset is broken and you actually have to actively disassemble one at a time to find the correct offset.  

Here's a real world example with mixed 2 and 4 byte instructions, and only 2 byte alignment:

0x0000000000000000:  08 46          mov  r0, r1
0x0000000000000002:  4F F6 FF 71    movw r1, #0xffff
0x0000000000000006:  01 46          mov  r1, r0

Fixing that is probably larger scope that this PR and I think this change likely doesn't block anything that wasn't already broken before, so it seems like good progress.

Note that for 64-bit ARM this is much easier, as instructions are always 4 bytes in length and 4 byte aligned.